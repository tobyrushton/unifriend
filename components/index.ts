export * from './providers'
export * from './screens'
