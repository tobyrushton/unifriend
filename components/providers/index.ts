export * from './LoadingProvider'
export * from './UserProvider'
export * from './NotificationProvider'
