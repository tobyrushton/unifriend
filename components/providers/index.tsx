export * from './LoadingProvider'
export * from './UserProvider'