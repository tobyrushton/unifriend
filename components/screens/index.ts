export * from './LoadingScreen'
export * from './AuthScreen'
export * from './ConfirmEmailScreen'
export * from './ForgottenPasswordScreen'
