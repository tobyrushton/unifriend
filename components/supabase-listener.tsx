'use client'

import { useRouter } from 'next/navigation'
import { useEffect, FC } from 'react'
import { supabase } from '../lib/supabase'

export const SupabaseListener: FC<{ accessToken?: string }> = ({
    accessToken,
}) => {
    const router = useRouter()

    useEffect(() => {
        supabase.auth.onAuthStateChange((_event, session) => {
            if (session?.access_token !== accessToken) {
                router.refresh()
            }
        })
    }, [accessToken, router])

    return null
}
