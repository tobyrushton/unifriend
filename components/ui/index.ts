'use client'

export * from './Logo'
export * from './Text'
export * from './Button'
export * from './Input'
export * from './Exit'
export * from './Notification'
export * from './Navbar'
export * from './DropDown'
export * from './ProfilePicture'
export * from './Toggle'
export * from './Message'
export * from './User'
