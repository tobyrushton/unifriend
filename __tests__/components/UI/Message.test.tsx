import { render } from '@testing-library/react'
import { Message } from '../../../components'

describe('Message', () => {
    it('should render correctly', () => {
        const { container } = render(<Message>Test</Message>)
        expect(container).toMatchSnapshot()
    })

    it('should render correctly with recieved prop', () => {
        const { container } = render(<Message recieved>Test</Message>)
        expect(container).toMatchSnapshot()
    })
})
