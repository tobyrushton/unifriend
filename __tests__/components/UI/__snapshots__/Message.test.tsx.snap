// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly 1`] = `
<div>
  <div
    class="message "
  >
    <p
      class="fontText   left"
    >
      Test
    </p>
  </div>
</div>
`;

exports[`Message should render correctly with recieved prop 1`] = `
<div>
  <div
    class="message recieved"
  >
    <p
      class="fontText   left"
    >
      Test
    </p>
  </div>
</div>
`;
